src/app/core/user/user.service.ts

```diff
@@ -14,7 +14,7 @@ export class UserService {
   userinfo: Userinfo;
 
   // tslint:disable-next-line: variable-name
-  private _userDoc = new BehaviorSubject(null);
```

Comment:
The toJSON() of null error, this is the real bug fix here: I switched from BehaviorSubject to ReplaySubject because the _userDoc should not have an initial value. The userDoc Observable shouldn't emit until the user has been loaded.